src/Propulsion.Kafka/Consumers.fs

```diff
@@ -330,6 +330,24 @@ module Core =
     let parseMessageKey defaultCategory = function
         | null -> FsCodec.StreamName.create defaultCategory ""
         | key -> Propulsion.Streams.StreamName.parseWithDefaultCategory defaultCategory key
+    let toTimelineEvent toDataAndContext (result : ConsumeResult<string, string>, index) =
+        let data, context = toDataAndContext result
+        FsCodec.Core.TimelineEvent.Create(index, String.Empty, data, context = context)
+
+    let toStreamName defaultCategory (result : ConsumeResult<string, string>) =
+        let m = Binding.message result
+        if m = null then invalidOp "Cannot dereference null message"
+        parseMessageKey defaultCategory m.Key
+
+    // This type is not a formal part of the API; the intent is to provide context as to the origin of the event for insertion into a parser error message
+    // Arguably it should be an anonymous record...
+    type ConsumeResultContext = { topic : string; partition : int; offset : int64 }
+    let toDataAndContext (result : ConsumeResult<string, string>) : byte[] * obj =
+        let m = Binding.message result
+        if m = null then invalidOp "Cannot dereference null message"
+        let data = System.Text.Encoding.UTF8.GetBytes m.Value
+        let context = { topic = result.Topic; partition = Binding.partitionValue result.Partition; offset = Binding.offsetValue result.Offset }
```

Comment:
Where `Binding` come from?

Reply:
https://github.com/jet/FsKafka/blob/master/src/FsKafka/FsKafka.fs#L13 and https://github.com/jet/FsKafka/blob/master/src/FsKafka0/FsKafka.fs#L13